e2e/tests/regression/test_replica.robot

```diff
@@ -0,0 +1,49 @@
+*** Settings ***
+Documentation    Replica Test Cases
+
+Test Tags    regression
+
+Resource    ../keywords/common.resource
+Resource    ../keywords/volume.resource
+Resource    ../keywords/setting.resource
+Resource    ../keywords/deployment.resource
+Resource    ../keywords/persistentvolumeclaim.resource
+Resource    ../keywords/workload.resource
+
+Test Setup    Set test environment
+Test Teardown    Cleanup test resources
+
+*** Variables ***
+${LOOP_COUNT}    1
+${RETRY_COUNT}    300
+${RETRY_INTERVAL}    1
+${DATA_ENGINE}    v1
+
+*** Test Cases ***
+Test Replica Rebuilding Per Volume Limit
+    [Tags]    coretest
+    [Documentation]    Test the volume always only have one replica scheduled for rebuild
+    ...
+    ...    1. Set soft anti-affinity to `true`.
+    ...    2. Create a volume with 1 replica.
+    ...    3. Attach the volume and write a few hundreds MB data to it.
+    ...    4. Scale the volume replica to 5.
+    ...    5. Monitor the volume replica list to make sure there should be only 1 replica in WO state.
+    ...    6. Wait for the volume to complete rebuilding. Then remove 4 of the 5 replicas.
+    ...    7. Monitoring the volume replica list again.
+    ...    8. Once the rebuild was completed again, verify the data checksum.
+    Given Set setting replica-soft-anti-affinity to true
+    And Create volume 0 with    numberOfReplicas=3    dataEngine=${DATA_ENGINE}
+    And Attach volume 0
+    And Wait for volume 0 healthy
+    And Write data to volume 0
+
+    When Update volume 0 replica count to 5
+    Then Only one replica rebuilding will start at a time for volume 0
+    And Wait until volume 0 replicas rebuilding completed
```

Comment:
Is it possible to change the keyword `Wait until volume 0 replicas rebuilding is completed` to emphasize that it ensures only one replica in the rebuild (WO) state until the rebuild is completed, for better match test case name `Test Replica Rebuilding Per Volume Limit`? Beside this, no further questions. Thank you.

Reply:
@chriscchien Could you suggest an appropriate name?

Reply:
How about `Only one replica for volume 0 rebuilds at a time until the rebuild is completed` for below two?
```robot
 Then Only one replica rebuilding will start at a time for volume 0
 And Wait until volume 0 replicas rebuilding completed
```

Reply:
@chriscchien How about adding another step `Monitor only one replica rebuilding will start at a time for volume 0`:

```
Then Only one replica rebuilding will start at a time for volume 0
And Monitor only one replica rebuilding will start at a time for volume 0
And Wait until volume 0 replicas rebuilding completed
```

This aligns with the description in the test skeleton.